llvm/include/llvm/Analysis/IndirectCallVisitor.h

```diff
@@ -12,27 +12,96 @@
 #ifndef LLVM_ANALYSIS_INDIRECTCALLVISITOR_H
 #define LLVM_ANALYSIS_INDIRECTCALLVISITOR_H
 
+#include "llvm/ADT/SetVector.h"
 #include "llvm/IR/InstVisitor.h"
 #include <vector>
 
 namespace llvm {
-// Visitor class that finds all indirect call.
+// Visitor class that finds indirect calls or instructions that gives vtable
+// value, depending on Type.
 struct PGOIndirectCallVisitor : public InstVisitor<PGOIndirectCallVisitor> {
+  enum class InstructionType {
+    kIndirectCall = 0,
+    kVTableVal = 1,
+  };
   std::vector<CallBase *> IndirectCalls;
-  PGOIndirectCallVisitor() = default;
+  SetVector<Instruction *, std::vector<Instruction *>> VTableAddrs;
```

Comment:
Add description for the new variable.

Reply:
Updated the call visitor to use heuristics; this way no need to require `-fwhole-program-vtables` (and thereby `-flto`) at instrumentation build.